.github/workflows/python-min-deps.yml

```diff
@@ -0,0 +1,172 @@
+name: Python Dependency Minimum Versions
+
+on:
+  push:
+    branches:
+      - "develop"
+      - "feature/**"
+  pull_request:
+    types: [opened, synchronize, reopened]
+  # Allows workflow to be called from other workflows
+  workflow_call:
+    inputs:
+      ref:
+        required: true
+        type: string
+      force-canary:
+        description: |
+          Forces the current build to be canary.
+          Canary builds test all Python versions and do not use constraints.
+        default: false
+        type: boolean
+      constraints-branch:
+        description: "The name of the branch from which the constraints files will be downloaded or compared with."
+        default: "constraints-develop"
+        type: string
+    secrets:
+      PARAMETER_PASSWORD:
+        description: "Token passed from caller workflows for snowflake integration tests"
+        required: true
+
+# Avoid duplicate workflows on same branch
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}-python
+  cancel-in-progress: true
+
+defaults:
+  run:
+    shell: bash --login -eo pipefail {0}
+
+env:
+  FORCE_COLOR: "1"
+
+jobs:
+  build_info:
+    runs-on: ubuntu-latest
+
+    name: "Build info"
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref }}
+          persist-credentials: false
+          submodules: "recursive"
+          fetch-depth: 2
+      - name: Set Python version vars
+        id: build_info
+        uses: ./.github/actions/build_info
+        with:
+          force-canary: ${{ inputs.force-canary || false }}
+
+    outputs:
+      PYTHON_MIN_VERSION: ${{ steps.build_info.outputs.PYTHON_MIN_VERSION }}
+
+  py_version:
+    needs:
+      - build_info
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+
+    # TODO: Should we add developer-friendly name for this job also?
+    #   This will unfortunately require a branch protection update.
+
+    env:
+      PYTHON_VERSION: "${{needs.build_info.outputs.PYTHON_MIN_VERSION}}"
+
+    steps:
+      - name: Checkout Streamlit code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.ref }}
+          persist-credentials: false
+          submodules: "recursive"
+
+      - name: Set up Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      # This section is an inlined copy of the make_init action
+      # Both because we need to do some parts of it differently
+      # and because it running in a separate action causes the python path
+      # setup to work weirdly, so some tests fail unless you run an install
+      # command in the workflow directly.
```

Comment:
You may have already experimented with this, but just to confirm - adding a new input flag to the `make_init` action to handle the differences below doesn't work?

- creating the Python environment cache key (lib/min-constraints-gen.txt vs. lib/dev-requirements.txt)
- restore virtualenv from cache (changed key name)
- Create Virtual Env (make python-init-test-min-deps vs. make python-init)

Reply:
I didn't try doing it as a new input flag, because it felt like it would result in a bad abstraction. While prototyping this, I made changes to `make_init` directly, and then when I discovered that the path only gets set up correctly if you do an additional install command in the workflow (the existing test workflows do `make develop` in a step after the `make_init` action), I decided to inline the action's steps.